Assets/00_Content/Scripts/Interactables/Beers/BarrelDropOff.cs

```diff
@@ -0,0 +1,52 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace Interactables.Beers {
+	public class BarrelDropOff : Interactable {
+		[SerializeField] private Image progressBarImage;
+		[SerializeField] private GameObject progressBar;
+		[SerializeField] private BeerTap beerTap;
+		[Tooltip("Timespan in s it takes to switch beerbarrels.")]
+		[SerializeField] private float switchTime = 2f;
+
+		private float switchingProgress = 0f;
+		private bool isHolding;
+
+		public override bool CanInteract(GameObject player, PickUp item) {
+			return item is BeerBarrel && beerTap.BeerAmount < beerTap.MaxBeerAmount;
+		}
+
+		public override void Interact(GameObject player, PickUp barrel) {
+			isHolding = true;
+			StartCoroutine(SwitchBeerTapBarrel(barrel));
+		}
+
+		public override void CancelInteraction(GameObject player, PickUp item) {
+			isHolding = false;
+		}
+
+		private IEnumerator SwitchBeerTapBarrel(PickUp barrel) {
+
+			while (switchingProgress <= switchTime && isHolding) {
+
+				switchingProgress += Time.deltaTime;
+
+				if (!progressBar.activeInHierarchy) progressBar.SetActive(true);
+
+				progressBarImage.fillAmount = switchingProgress / switchTime;
+
+				if (switchingProgress > switchTime) {
+					beerTap.BeerAmount = beerTap.MaxBeerAmount;
```

Comment:
Nothing breaking but couldn't this be moved to the beertap as a refill function to lower the "knowledge" required to refill a tap?

Reply:
I guess I could do that yeah, but I'm sorry but I don't understand why that would be better. Wouldn't it result in the same amount of code, just some of it moved to the beerTap?
I'm guessing there is some principle or advantage to this that I'm not seeing, could you explain it?